File: Python/1582_numSpecial.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        mOnes = [0] * len(mat)
        nOnes = [0] * len(mat[0])

        for i in range(len(mat)):  
            for j in range(len(mat[0])):
                if mat[i][j] == 1:
                    mOnes[i] += 1
                    nOnes[j] +=1

        count = 0
        for i in range(len(mat)):  
            for j in range(len(mat[0])):
                if mat[i][j] == 1 and mOnes[i] == 1 and nOnes[j] == 1:
                    count+=1
        return count

File: C++/1582_numSpecial.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        vector<int> mOnes(mat.size(), 0);
        vector<int> nOnes(mat[0].size(), 0);

        for(int i = 0; i < mat.size(); i++){  
            for(int j = 0; j < mat[0].size(); j++){
                if(mat[i][j] == 1){
                    mOnes[i] += 1;
                    nOnes[j] +=1;
                }
            }
        }
        int count = 0;
        for(int i = 0; i < mat.size(); i++){  
            for(int j = 0; j < mat[0].size(); j++){
                if(mat[i][j] == 1 && mOnes[i] == 1 && nOnes[j] == 1){
                    count++;
                }
            }
        }
        return count;
    }
};
